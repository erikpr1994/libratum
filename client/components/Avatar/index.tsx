import { useContext } from 'react';
import { loginContext } from '@hooks/useLogin';

type AvatarType = {
  alt: string;
  src?: string;
  text?: string;
  height: number;
  width: number;
};

export default function Avatar({ alt, src, text, height, width }: AvatarType) {
  // TODO: Fix context bug -> logged is set as undefined when clicked on login
  const { logged } = useContext(loginContext);

  if (!src) src = 'placeholder/avatar.png';
  return (
    <>
      {logged && (
        <div>
          {text && <strong>{text}</strong>}
          <img alt={alt} src={src} title={alt} />
        </div>
      )}
      <style jsx>{`
        div {
          display: flex;
          align-items: center;
          justify-content: flex-end;
          height: 100%;
        }

        strong {
          margin-right: 8px;
        }

        img {
          border-radius: 9999px;
          height: ${height}px;
          width: ${width}px;
        }
      `}</style>
    </>
  );
}
