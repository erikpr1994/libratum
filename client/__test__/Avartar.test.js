import React from 'react';
import { cleanup, render, screen } from '@testing-library/react';

import Avatar from '../components/Avatar';

const avatarComponent = (AltProfilePic, height, width, text) =>
  render(
    <Avatar alt={AltProfilePic} height={height} width={width} text={text} />
  );

describe('Avatar component', () => {
  afterEach(() => {
    cleanup();
  });

  test('render correctly', () => {
    const testComponent = avatarComponent('alt', 50, 50);
    expect(testComponent).toMatchSnapshot();
  });

  test('should show text', () => {
    avatarComponent('alt', 50, 50, 'hello');
    expect(screen.findByText(/hello/i)).toBeTruthy();
  });
});
