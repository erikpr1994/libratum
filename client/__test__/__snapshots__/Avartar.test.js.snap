// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component render correctly 1`] = `
<body>
  <div />
</body>
`;
